doc/directory-structure.md

```diff
@@ -0,0 +1,212 @@
+OpenSSL Repository Structure
+===========================
+
+This document provides an overview of the OpenSSL repository's directory
+structure, explaining what the major directories and files are, their purpose,
+and how they relate to each other.
+
+This documentation is available in multiple locations:
+
+* As a man page when OpenSSL is installed on your system
+* As a web page on OpenSSL's official website (www.openssl.org)
```

Comment:
This document will not be a man page or a web page (only the .pod files get automatically made into these)